### ISSUE TITLE
An Arpeggiator

### ISSUE THREAD

**Issue by blackearth2014** [mschack/VCV-Rack-Plugins#20]
msHack,

I had just thought of two more implementations or devices.  Can you develop a new 6x32 Trigger Sequencer to trigger 8, 16 and 32 notes or pulse to get those flanging kicks, snares and hats EDM drops. This would require to make a larger graph for use to see the grids.

Also an Arpeggiator will be nice with/without internal clock with quantization abilities.

**Comment by mschack** [comment]
so like my current 6x32 except lengthen or shorten the note duration instead of output a level?

**Comment by blackearth2014** [comment]
Yes. Its like having grid with smaller time values. See attached image from a 909 drum machine.

![909drmmachinegrid](https://user-images.githubusercontent.com/7956901/33221492-de98a86c-d11d-11e7-9f7d-3720992cabf4.png)

**Comment by mschack** [comment]
I should be using this thread I guess... Any ref vids?

**Comment by blackearth2014** [comment]
I will post some that I think are unique and feature rich.

This is your season
Omar Brown
Blessed House Media

On Nov 24, 2017, at 5:35 PM, mschack <notifications@github.com> wrote:

I should be using this thread I guess... Any ref vids?

**Comment by blackearth2014** [comment]
http://www.youtube.com/watch?v=19i1mfvkg5c is a Apple Logic arpeggiator.  I know that you can't do polyphony stuff write now, but I was thinking that you can combine the triad keyboard with the 6x32 trigger sequencer would make a wicked arp. Here is more details. The program change options can hold a map of a chord. For instance, the keyboard widgets now only allow one keypress at a time. What if the keyboard widget allows 3 to 4 simultaneous key presses. The Arp could recognize the notes and cycle through them. This way we can write chords and assigned them to different program slots. Then the master clock can drive the pattern changes and it advances forward to the next chord and pattern. Just how your sequencers are working now. All you need is one nice size arp to control the VCO's or dual rows for the arp module.

**Comment by blackearth2014** [comment]
I had just thought of this. Could you add a counter feature to allow you to set the number of steps or pulses to reach before it goes to the next pattern and/or program change? Better yet, add a port to the Arp that will allow you to connect an external counter for added flexibility. You may have to develop a counter though. I find the counter to be extremely useful. You can hold on to a section of music until you are ready for it to change. Some like intro, verse, chorus and etc. Because copying and pasting of patterns can get old quit.

**Comment by mschack** [comment]
So type in multiple keys on the keyboard then have modes like up, down, or up/down to continuously cycle through the chord given an input trigger?

**Comment by blackearth2014** [comment]
You got it.
